apps/schools/domains/common/components/containers/BaseLayoutComponents/OrganizationSelect/index.tsx

```diff
@@ -24,8 +24,10 @@ export const OrganizationSelect: React.FC<OrganizationSelectProps> = (props) =>
     const ORGANIZATION_SELECT_SHOW_ACTIONS: SelectProps<string>['showAction'] = ['focus', 'click']
 
     const organizationCookieChange = (value: string) => {
-        setOrganizationId(value)
-        window.location.reload()
+        refetch()
+        setTimeout(() => {
+            setOrganizationId(value)
+        }, 1000)
```

Comment:
why you need setTimeout here
value now is not undefined it should work good?
maybe you can delete refetch because organizationProvider make request for you

Reply:
organizationProvider does not make a request when creating an organization

Reply:
value now is not undefined.

Reply:
setTimeout is needed here so that prefetch() has time to execute